client/context.go

```diff
@@ -280,7 +281,13 @@ func (ctx Context) WithInterfaceRegistry(interfaceRegistry codectypes.InterfaceR
 // client-side config from the config file.
 func (ctx Context) WithViper(prefix string) Context {
 	v := viper.New()
+
+	if prefix == "" {
+		prefix, _ = os.Executable()
+	}
+
 	v.SetEnvPrefix(prefix)
+	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_"))
```

Comment:
Could you please explain why this replacer?

Reply:
For consistency with server: https://github.com/cosmos/cosmos-sdk/blob/main/server/util.go#L151